src/geojson-layer.ts

```diff
@@ -300,7 +300,19 @@ export default class GeoJSONLayer extends React.Component<Props, {}> {
           }
         }
       });
+
+      if (before !== props.before) {
+        map.moveLayer(layerId, props.before);
+      }
+
     });
+
```

Comment:
you can remove from line 309 to 315, what you do line 304 to 306 is the same as it is included in a forEach which run for every layer types. See line 263. Once this is done I will merge this PR

Reply:
I'd rather remove line 304-306 which I added in a previous commit and forgot to remove later. Reason is, I can not be sure about order of keys that will come from `Object.keys(typeToLayerLUT)` as per definition this is an unordered collection of properties. So I listed the keys explicitly in line 311 to make sure I preserve order of geojson sublayers.
What do you think?

Reply:
I did some changes regarding the `typeToLayerLUT` object. Thanks for your help 👍 I kept your commits
cf: https://github.com/alex3165/react-mapbox-gl/pull/438

Reply:
Great! Nice solution.